clients/wallet/src/resubmissions.rs

```diff
@@ -0,0 +1,795 @@
+use std::sync::Arc;
+
+use crate::{
+	error::{
+		CacheError, CacheErrorKind, Error,
+		Error::{DecodeError, ResubmissionError},
+	},
+	StellarWallet, TransactionResponse,
+};
+use primitives::{
+	stellar::{Memo, Transaction, TransactionEnvelope, XdrCodec},
+	TransactionEnvelopeExt,
+};
+use std::time::Duration;
+use tokio::time::sleep;
+
+#[cfg(test)]
+use mocktopus::macros::mockable;
+use tokio::sync::RwLock;
+
+pub type FailedTasks =
+	Arc<RwLock<Vec<tokio::sync::oneshot::Receiver<Option<TransactionEnvelope>>>>>;
+
+#[cfg_attr(test, mockable)]
+impl StellarWallet {
+	/// Returns a list of receivers that will receive a `TransactionEnvelope`
+	/// if resubmission failed
+	///
+	/// # Arguments
+	/// * `auto_check` - creates another thread that does the checking for failed resubmissions
+	pub async fn resubmit_transactions_from_cache(&self, auto_check: bool) -> FailedTasks {
+		// a collection of all tasks currently running
+		let failed_tasks = Arc::new(RwLock::new(vec![]));
+
+		// perform the resubmission
+		self._resubmit_transactions_from_cache(failed_tasks.clone(), auto_check).await;
+
+		// clone the task, to share this in another thread
+		let failed_tasks_clone = failed_tasks.clone();
+
+		// clone self, to be able to use this in another thread
+		let me = Arc::new(self.clone());
+		// spawn a thread to resubmit envelopes from cache
+		tokio::spawn(async move {
+			let me_clone = Arc::clone(&me);
+			loop {
+				// loops every 30 minutes or 1800 seconds
+				pause_process_in_secs(1800).await;
```

Comment:
Nit: We should probably use a proper constant here instead of `1800` as a magic number.